File: library/src/main/java/org/incendo/interfaces/element/StaticElement.java
package org.incendo.interfaces.element;

import org.incendo.interfaces.click.ClickHandler;
import org.incendo.interfaces.drawable.Drawable;

public record StaticElement(Drawable drawable, ClickHandler clickHandler) implements Element {

    public StaticElement(final Drawable drawable) {
        this(drawable, ClickHandler.EMPTY);
    }

}

File: next/src/main/kotlin/org/incendo/interfaces/next/element/StaticElement.kt
package org.incendo.interfaces.next.element

import org.incendo.interfaces.next.click.ClickHandler
import org.incendo.interfaces.next.drawable.Drawable

public class StaticElement public constructor(
    private val drawable: Drawable,
    private val clickHandler: ClickHandler = ClickHandler.EMPTY,
) : Element {

    override fun drawable(): Drawable = drawable

    override fun clickHandler(): ClickHandler = clickHandler
}
